micrometer-core/src/main/java/io/micrometer/core/instrument/binder/grpc/DefaultGrpcClientObservationConvention.java

```diff
@@ -43,14 +39,15 @@ public String getContextualName(GrpcClientObservationContext context) {
 
     @Override
     public KeyValues getLowCardinalityKeyValues(GrpcClientObservationContext context) {
-        List<KeyValue> keyValues = new ArrayList<>();
-        keyValues.add(LowCardinalityKeyNames.METHOD.withValue(context.getMethodName()));
-        keyValues.add(LowCardinalityKeyNames.SERVICE.withValue(context.getServiceName()));
-        keyValues.add(LowCardinalityKeyNames.METHOD_TYPE.withValue(context.getMethodType().name()));
-        if (context.getStatusCode() != null) {
-            keyValues.add(LowCardinalityKeyNames.STATUS_CODE.withValue(context.getStatusCode().name()));
-        }
-        return KeyValues.of(keyValues);
+        String statusCode = context.getStatusCode() != null ? context.getStatusCode().name() : UNKNOWN;
+        String peerPort = context.getPeerPort() != null ? context.getPeerPort().toString() : UNKNOWN;
+        return KeyValues.of(LowCardinalityKeyNames.METHOD.withValue(context.getMethodName()),
+                LowCardinalityKeyNames.SERVICE.withValue(context.getServiceName()),
+                LowCardinalityKeyNames.SYSTEM.withValue("grpc"),
+                LowCardinalityKeyNames.METHOD_TYPE.withValue(context.getMethodType().name()),
+                LowCardinalityKeyNames.STATUS_CODE.withValue(statusCode),
+                LowCardinalityKeyNames.PEER_NAME.withValue(context.getPeerName()),
```

Comment:
I think we may have talked about this, but unfortunately I've forgotten - should peer name be low cardinality or high cardinality?

Reply:
@ttddyy What do you think?

Reply:
I think it is a low cardinality. The peername is essentially a hostname of the remote call target.
Different from making http calls to external network web servers, usually remote call targets should be a fixed number of servers with fixed ports within the network.
So, I don't expect the number of the target to be many.